pkg/dmaasbackup/periodic_backup.go

```diff
@@ -173,3 +180,30 @@ func (d *dmaasBackup) cleanupPeriodicSchedule(dbkp *v1alpha1.DMaaSBackup) error
 	}
 	return nil
 }
+
+// getSuccessfulScheduleIdx return index for the veleroschedules having successful backup for the given dmaasbackup
+// if no successful backup exists then it returns error
+func (d *dmaasBackup) getSuccessfulScheduleIdx(dbkp *v1alpha1.DMaaSBackup) (int, error) {
+	for idx, schedule := range dbkp.Status.VeleroSchedules {
+		if schedule.Status == v1alpha1.Erased {
```

Comment:
should we do 'continue' for empty schedule.Status

Reply:
We can add it, but it won't be help here, since we are keeping empty status only if schedule is reserved, and not created.